### DISCUSSION TITLE
Connect to Twitch chat in rirc

### DISCUSSION THREAD

**Discussion by daviewales** [rcr/rirc#35] [Show and tell]
I wrote a blog post about [connecting to Twitch chat using rirc](https://daviewales.com/twitch/irc/cli/2021/07/05/twitch-chat-in-rirc-irc-client.html).

Here is the TL:DR:

1. Get an authentication code from [here](https://twitchapps.com/tmi/). Use this instead of a password.
2. Connect using rirc:
    ```
    ./rirc -s irc.chat.twitch.tv -p 6697 -w <authentication_token> -n <username>
    ```
3. Join channel using the IRC join command:
    ```
    /join #channel
    ```

**Comment by rcr** [comment]
This is great, I didn't know twitch supported irc. I'm going to check this out and see what liberties they take with the protocol.

I'm getting set to merge dev to master some time this week so the build instructions will change slightly as I've gotten rid of submodules, will update this discussion when I do so.

Thanks for checking out the project!

**Comment by daviewales** [comment]
These are the [Twitch Chat IRC docs](https://dev.twitch.tv/docs/irc/guide).

**Comment by rcr** [comment]
The build instructions have slightly changed with v0.1.5 -- git sub modules have been removed.
